src/query/api/v1/handler/prometheus/handleroptions/fetch_options.go

```diff
@@ -58,6 +61,15 @@ type FetchOptionsBuilder interface {
 type FetchOptionsBuilderOptions struct {
 	Limits        FetchOptionsBuilderLimitsOptions
 	RestrictByTag *storage.RestrictByTag
+	Timeout       time.Duration
+}
+
+// Validate validates the fetch options builder options.
+func (o FetchOptionsBuilderOptions) Validate() error {
+	if o.Timeout <= 0 {
```

Comment:
nit; may be better to do the `o.Timeout > 0` and `o.Timeout < maxTimeout` checks in the same place?